napari/layers/labels/labels.py

```diff
@@ -388,14 +377,43 @@ def properties(self) -> Dict[str, np.ndarray]:
 
     @properties.setter
     def properties(self, properties: Dict[str, np.ndarray]):
-        if not isinstance(properties, dict):
-            properties, label_index = dataframe_to_properties(properties)
-            if label_index is None:
-                label_index = self._map_index(properties)
-            self._label_index = label_index
-        self._properties = self._validate_properties(properties)
+        self._properties, self._label_index = self._prepare_properties(
+            properties
+        )
         self.events.properties()
 
+    @classmethod
+    def _prepare_properties(cls, properties) -> Tuple[dict, dict]:
+        """
+        Parameters:
```

Comment:
```suggestion
        """Convert an input properties value to a standard dict-of-columns format.

        Parameters
        ----------
```

Reply:
Probably this should be a function rather than a method, but it is not urgent to fix this in this PR.

Reply:
this call other classmethod/staticmethod `cls._validate_properties` so need to be classmethod